src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetSecureRandomCommand.cs

```diff
@@ -0,0 +1,716 @@
+                        {
+                            Debug.Assert(_chosenListItems.Count == _numberOfProcessedListItems, "Initial K elements should all be included in chosenListItems");
+                            _chosenListItems.Add(item);
+                        }
+                        else
+                        {
+                            Debug.Assert(_chosenListItems.Count == Count, "After processing K initial elements, the length of chosenItems should stay equal to K");
+
+                            // (1)
+                            if (Generator.Next(_numberOfProcessedListItems + 1) < Count)
+                            {
+                                // (2)
+                                int indexToReplace = Generator.Next(_chosenListItems.Count);
+                                _chosenListItems[indexToReplace] = item;
+                            }
+                        }
+
+                        _numberOfProcessedListItems++;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// This method implements the EndProcessing method for Get-SecureRandom command.
+        /// </summary>
+        protected override void EndProcessing()
+        {
+            if (EffectiveParameterSet == MyParameterSet.RandomListItem)
+            {
+                // make sure the order is truly random
+                // (all permutations with the same probability)
+                // O(n) time
+                int n = _chosenListItems.Count;
+                for (int i = 0; i < n; i++)
+                {
+                    // randomly choose j from [i...n)
+                    int j = Generator.Next(i, n);
+
+                    WriteObject(_chosenListItems[j]);
+
+                    // remove the output object from consideration in the next iteration.
+                    if (i != j)
+                    {
+                        _chosenListItems[j] = _chosenListItems[i];
+                    }
+                }
+            }
+        }
+
+        #endregion Processing methods
+    }
+
+    /// <summary>
+    /// Provides an adapter API for random numbers that may be either cryptographically random, or
+    /// generated with the regular pseudo-random number generator. Re-implementations of
+    /// methods using the NextBytes() primitive based on the CLR implementation:
+    ///     https://referencesource.microsoft.com/#mscorlib/system/random.cs.
+    /// </summary>
+    internal class PolymorphicRandomNumberGenerator
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PolymorphicRandomNumberGenerator"/> class.
+        /// </summary>
+        public PolymorphicRandomNumberGenerator()
+        {
+            _cryptographicGenerator = RandomNumberGenerator.Create();
+            _pseudoGenerator = null;
+        }
+
+        internal PolymorphicRandomNumberGenerator(int seed)
```

Comment:
Add a comment that this leads to non-cryptographically secure random number?

Reply:
adding comment